code/src/components/TodoItem.js

```diff
@@ -0,0 +1,54 @@
+import React, { useEffect, useState } from "react";
+import { useDispatch, useSelector } from "react-redux";
+import { todos } from "../reducers/todos.js";
+import moment from 'moment'
+import { CustomCheckbox } from './CustomCheckbox';
+
+
+export const TodoItem = ({ itemIndex }) => {
+  const item = useSelector((store) => store.todos.list.items[itemIndex]);
+  const dispatch = useDispatch();
+
+  const handleCheckboxClick = (e) => {
+    dispatch(todos.actions.setDone({
+      itemIndex: itemIndex,
+      done: !item.done,
+    })
+    );
+  };
```

Comment:
Heads up!
When you're using the index to indicate what item should be checked or not, there's a bug appearing.
Try to add a bunch of todos, check the first one but not the second one, the todo with index 0 will be checked. Then remove the first one.
You'll see that now the second one will be checked instead. This is because that one now gets the "new" index 0.

Reply:
To fix this, try to use an `id` instead of an index to deal with the done state.